torchci/pages/sli.tsx

```diff
@@ -218,67 +221,153 @@ function WorkerTypePicker({
 }
 
 export default function Page() {
-  const [startTime, setStartTime] = useState(dayjs().subtract(1, "week"));
-  const [stopTime, setStopTime] = useState(dayjs());
-  const [timeRange, setTimeRange] = useState<number>(7);
-  const [workerTypes, setWorkerTypes] = useState<string[]>(["all"]);
+  const router = useRouter();
 
-  const timeParams: { [key: string]: any } = {
-    startTime: startTime.utc().format("YYYY-MM-DDTHH:mm:ss.SSS"),
-    stopTime: stopTime.utc().format("YYYY-MM-DDTHH:mm:ss.SSS"),
-  };
+  const { query } = router;
 
-  const [ttsPercentile, setTtsPercentile] = useState<string>("p95");
+  const initialStopTime = query.stopTime
+    ? dayjs(query.stopTime as string)
+    : dayjs();
+  const initialTimeRange = query.timeRange
+    ? parseInt(query.timeRange as string)
+    : 7;
+  const initialStartTime = query.startTime
+    ? dayjs(query.startTime as string)
+    : initialStopTime.subtract(initialTimeRange, "day");
+  const initialWorkerTypes = query.workerTypes
+    ? (query.workerTypes as string).split(",")
+    : ["pet", "ephemeral", "nonephemeral"];
+  const initialTtsPercentile = query.ttsPercentile
+    ? (query.ttsPercentile as string)
+    : "p80";
+
+  const [startTime, setStartTime] = useState(initialStartTime);
+  const [stopTime, setStopTime] = useState(initialStopTime);
+  const [timeRange, setTimeRange] = useState(initialTimeRange);
+  const [workerTypes, setWorkerTypes] = useState(initialWorkerTypes);
+  const [ttsPercentile, setTtsPercentile] = useState(initialTtsPercentile);
+  const [routerReady, setRouterReady] = useState(false);
+
+  if (!routerReady && router.isReady) {
+    setRouterReady(true);
+    setStartTime(initialStartTime);
+    setStopTime(initialStopTime);
+    setTimeRange(initialTimeRange);
+    setWorkerTypes(initialWorkerTypes);
+    setTtsPercentile(initialTtsPercentile);
+  }
+
+  const fullUrl = routerReady
+    ? `${window.location.origin}${router.asPath}`
+    : "";
+
+  useEffect(() => {
+    if (!router.isReady) return;
+
+    const params = new URLSearchParams();
+
+    if (timeRange !== -1) {
+      params.set("timeRange", timeRange.toString());
+    } else if (startTime && stopTime) {
+      params.set("startTime", startTime.utc().format("YYYY-MM-DD"));
+      params.set("stopTime", stopTime.utc().format("YYYY-MM-DD"));
+    } else {
+      params.set("timeRange", "7");
+    }
+
+    if (workerTypes) {
+      params.set("workerTypes", workerTypes.join(","));
+    } else {
+      params.set("workerTypes", initialWorkerTypes.join(","));
+    }
```

Comment:
I'm bad a typescript, but aren't ternary a thing there?
```suggestion
    params.set("workerTypes", (workerTypes?:initialWorkerTypes).join(","));
```